src/Umbraco.Infrastructure/Persistence/Repositories/Implement/KeyValueRepository.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NPoco;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Persistence.Dtos;
+using Umbraco.Core.Persistence.Querying;
+using Umbraco.Core.Scoping;
+
+namespace Umbraco.Core.Persistence.Repositories.Implement
+{
+    internal class KeyValueRepository : NPocoRepositoryBase<string, IKeyValue>, IKeyValueRepository
+    {
+        public KeyValueRepository(IScopeAccessor scopeAccessor, ILogger logger)
+            : base(scopeAccessor, AppCaches.NoCache, logger)
+        { }
+
+        /// <summary>
+        /// Gets a value directly from the database, no scope, nothing.
+        /// </summary>
+        /// <remarks>Used by <see cref="Runtime.CoreRuntime"/> to determine the runtime state.</remarks>
+        internal static string GetValue(IUmbracoDatabase database, string key)
```

Comment:
I think this can be moved out to an extension method on IUmbracoDatabase. There's just something funky about this being an internal class and an internal method and that we don't want to really ever call a Repository class directly. Or maybe this can remain a static method here to return the `Sql` object used to do the query, that way all SQL production is still in a repository class. But the calling of this method from `RuntimeState` could then be `CurrentMigrationState = database.GetFromKeyValueTable(stateValueKey);` (or similar) and that method could call this method which could be changed to `GetKeyValueSql`. That way we can have the `database.GetFromKeyValueTable` as a public method in case it's needed elsewhere or by people.

Reply:
Agree that's better.  Have implemented.